sdk/eventhub/azure-eventhubs/azure/eventhub/common.py

```diff
@@ -181,6 +205,8 @@ def sequence_number(self):
 
         :rtype: int or long
         """
+        if self._need_further_parse:
+            self._parse_message_properties()
```

Comment:
uamqp already lazily loads properties. I guess another lazy loading won't improve much performance

Reply:
I'm afraid I had to do the lazy parse here.
because the parsing actually takes place when properties of message are assigned to events.
so we need make the assignment lazy too.

Reply:
I think we can potentially only put the lazy loading into uamqp... we can discuss offline.

Reply:
lazy parsing removed